lib/node.js

```diff
@@ -687,7 +698,28 @@ declare module "http" {
 }
 
 declare module "https" {
-  // TODO
+  declare class Server extends tls$Server {
+    listen(port: number, hostname?: string, backlog?: number, callback?: Function): Server;
+    listen(path: string, callback?: Function): Server;
+    listen(handle: Object, callback?: Function): Server;
+    close(callback?: Function): Server;
+    setTimeout(msecs: number, callback: Function): Server;
+    timeout: number;
+  }
+
+  declare class ClientRequest extends http$ClientRequest {}
+  declare class IncomingMessage extends http$IncomingMessage {}
+
+  declare function createServer(options: Object, secureConnectionListener?: function): typeof Server;
```

Comment:
Fixed. 

I was a little fuzzy on the meaning of `typeof` and did this by imitation of other places (e.g. `tls.Server.createServer`) that appear to be making the same error.  
(I'll try to make a pass to fix in a separate PR at some point).